cyclonedx/model/component.py

```diff
@@ -103,7 +103,7 @@ def __init__(self, name: str, component_type: ComponentType = ComponentType.LIBR
                  publisher: Optional[str] = None, group: Optional[str] = None, version: Optional[str] = None,
                  description: Optional[str] = None, scope: Optional[ComponentScope] = None,
                  hashes: Optional[List[HashType]] = None, licenses: Optional[List[LicenseChoice]] = None,
-                 copyright: Optional[str] = None, purl: Optional[PackageURL] = None,
+                 copyright: Optional[str] = None, purl: Optional[PackageURL] = None, cpe: Optional[str] = None,
```

Comment:
:x: 

adding a new parameter not at the end of a function/method definition is considered a  breaking change,
because a call via non-keyword-arguments (`foo(1,2, 3)` instead of `foo(1, bar=2, baz=3)`) would require changing of order of parameters downstream.

Therefore, please add new parameters to the end.

@madpah we could enforce that some parameters must be keyword-arguments (`kwargs`) by the following [PEP3102](https://www.python.org/dev/peps/pep-3102/). Enforcing keyword-arguments would be considered a breaking change, IMHO.

Reply:
closed via f2d47b6